src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/CombinationController.php

```diff
@@ -102,6 +105,96 @@ public function editAction(Request $request, int $combinationId): Response
+     * @param string $languageCode
+     *
+     * @return JsonResponse
+     */
+    public function searchCombinationProductsAction(
+        Request $request,
+        string $languageCode
+    ): JsonResponse {
+        $langRepository = $this->get('prestashop.core.admin.lang.repository');
+        $language = $langRepository->getOneByLocaleOrIsoCode($languageCode);
+        if (null === $language) {
+            return $this->json([
+                'message' => sprintf(
+                    'Invalid language code %s was used which matches no existing language in this shop.',
+                    $languageCode
+                ),
+            ], Response::HTTP_BAD_REQUEST);
+        }
+
+        $shopId = $this->get('prestashop.adapter.shop.context')->getContextShopID();
+        if (empty($shopId)) {
+            $shopId = $this->get('prestashop.adapter.legacy.configuration')->getInt('PS_SHOP_DEFAULT');
+        }
+
+        try {
+            /** @var CombinationForAssociation[] $combinationProducts */
+            $combinationProducts = $this->getQueryBus()->handle(new SearchCombinationsForAssociation(
+                $request->get('query', ''),
+                $language->getId(),
+                (int) $shopId,
+                (int) $request->get('limit', 20)
+            ));
+        } catch (ProductConstraintException $e) {
+            return $this->json([
+                'message' => $e->getMessage(),
+            ], Response::HTTP_BAD_REQUEST);
+        }
+
+        if (empty($combinationProducts)) {
+            return $this->json([], Response::HTTP_NOT_FOUND);
+        }
+
+        return $this->json(
+            $this->formatCombinationProductsForAssociation(
+                $combinationProducts,
+                new LanguageId($language->getId())
+            )
+        );
+    }
+
+    /**
+     * @param CombinationForAssociation[] $productsForAssociation
+     *
+     * @return array
+     */
+    private function formatCombinationProductsForAssociation(
+        array $productsForAssociation,
+        LanguageId $languageId
+    ): array {
+        $combinationNameBuilder = $this->get('prestashop.core.product.combination.name_builder.combination_name_builder');
+        $attributeRepository = $this->get('prestashop.adapter.attribute.repository.attribute_repository');
+        $productsData = [];
+        foreach ($productsForAssociation as $productForAssociation) {
+            $productName = $productForAssociation->getName();
+            if ($productForAssociation->getCombinationId() > 0) {
+                $attributesInformation = $attributeRepository->getAttributesInfoByCombinationIds(
+                    [new CombinationId($productForAssociation->getCombinationId())],
+                    $languageId
+                );
+                $productName .= ' : ' . $combinationNameBuilder->buildName(
+                        $attributesInformation[$productForAssociation->getCombinationId()]
+                    );
```

Comment:
Translation service is required here :
- in english language there is no space before the colon
- RTL display is inversed

Reply:
Indeed, same idea as in the pack details handler This makes me think that this name building should probably be in the search handler

And I would suggest the same kind of optimization as in the details handler Try and get all attribute information in a single query by getting the whole list of combination IDs and using it as a parameter for `getAttributesInfoByCombinationIds`

Reply:
This whole part should be done in the handler, this will avoid fetching the attributes more than once This will allow integrating the translator in the handler itself So the whole controller code will be simpler and cleaner And we can behat test the expected name building

Reply:
logic move into SearchCombinationsForAssociationHandler class as a private method